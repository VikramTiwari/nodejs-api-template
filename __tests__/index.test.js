// the goal is to only require this file and be sure all the services started
const httpServer = require('../src/index')
// mock axios to make API calls
const axios = jest.requireActual(`axios`)

const { PORT = 8080 } = process.env

describe(`Services`, () => {
  beforeAll(done => {
    httpServer.once(`listening`, done)
    if (httpServer.listening) {
      done()
    }
  }, 10 * 1000)
  afterAll(done => {
    httpServer.close(() => {
      setTimeout(done, 100)
    })
  }, 10 * 1000)

  describe(`Web server started`, () => {
    test(
      `Web server started - status`,
      done => {
        axios.get(`http://localhost:${PORT}/?env=test`).then(res => {
          expect(res).toMatchSnapshot()
          done()
        })
      },
      10 * 1000,
    )
  })
})
