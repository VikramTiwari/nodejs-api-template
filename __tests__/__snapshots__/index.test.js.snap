// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services Web server started Web server started - status 1`] = `
Object {
  "config": Object {
    "adapter": [Function],
    "data": undefined,
    "headers": Object {
      "Accept": "application/json, text/plain, */*",
    },
    "maxBodyLength": -1,
    "maxContentLength": -1,
    "method": "get",
    "timeout": 0,
    "transformRequest": Array [
      [Function],
    ],
    "transformResponse": Array [
      [Function],
    ],
    "url": "http://localhost:8080/?env=test",
    "validateStatus": [Function],
    "xsrfCookieName": "XSRF-TOKEN",
    "xsrfHeaderName": "X-XSRF-TOKEN",
  },
  "data": Object {
    "env": "test",
  },
  "headers": Object {
    "content-type": "application/json; charset=utf-8",
  },
  "request": XMLHttpRequest {},
  "status": 200,
  "statusText": "OK",
}
`;
